src/dk/aau/cs/ds306e18/tournament/ui/bracketObjects/DoubleEliminationNode.java

```diff
@@ -0,0 +1,171 @@
+        int rounds = doubleElimination.getUpperBracketRounds();
+        int m = 0; // match index
+        for (int r = 0; r < rounds; r++) {
+            int matchesInRound = pow2(r);
+            int column = rounds - 1 - r;
+            int rowSpan = pow2(column);
+
+            // Add matches for round r
+            for (int i = 0; i < matchesInRound; i++) {
+                Match match = upperBracket[m++];
+                VBox box = createMatchBox(match, rowSpan, false);
+                upperGrid.add(box, column, (matchesInRound - 1 - i) * rowSpan);
+            }
+        }
+
+        // Final match is added to upper bracket
+        VBox finalsBox = createMatchBox(doubleElimination.getFinalMatch(), pow2(rounds - 1), false);
+        upperGrid.add(finalsBox, rounds, 0);
+
+        // Extra match is only shown when needed
+        VBox extraBox = createMatchBox(doubleElimination.getExtraMatch(), pow2(rounds - 1), true);
+        upperGrid.add(extraBox, rounds + 1, 0);
+        updateDisplayOfExtraMatch();
+
+        // Lower bracket
+        Match[] lowerBracket = doubleElimination.getLowerBracket();
+        if (lowerBracket.length > 0) {
+            int matchesInCurrentRound = pow2(rounds - 2);
+            int column = 0;
+            int rowSpan = 1;
+            m = 0; // match index
+            while (true) {
+
+                for (int i = 0; i < matchesInCurrentRound; i++) {
+                    Match match = lowerBracket[m++];
+                    VBox box = createMatchBox(match, rowSpan, false);
+                    lowerGrid.add(box, column, i * rowSpan);
+                }
+
+                // Half the number of matches in a round every other round
+                if (column % 2 == 1) {
+                    if (matchesInCurrentRound == 1) {
+                        break; // We can't do more halving
+                    }
+                    matchesInCurrentRound /= 2;
+                    rowSpan *= 2;
+                }
+
+                column++;
+            }
+        }
+    }
+
+    private VBox createMatchBox(Match match, int rowSpan, boolean isExtra) {
+
+        VBox box = new VBox();
+
+        // Some matches are null because of byes. In those cases the VBox will just be empty
+        if (match != null) {
+            MatchVisualController mvc = boc.loadVisualMatch(match);
+            mvcs.add(mvc);
+            box.getChildren().add(mvc.getRoot());
+            mvc.setShowIdentifier(true);
+
+            if (isExtra) {
+                extraMatchMVC = mvc;
+            }
+        }
+
+        box.setAlignment(Pos.CENTER);
+        box.setMinHeight(CELL_HEIGHT * rowSpan);
+
+        GridPane.setRowSpan(box, rowSpan);
+        GridPane.setMargin(box, MARGINS);
+        GridPane.setValignment(box, VPos.CENTER);
+
+        return box;
+    }
+
+    /** Creates the line between upper and lower bracket */
+    private Node getLine() {
+        Line line = new Line(0, 0, 1000, 0);
```

Comment:
Having the width of the horizontal divider set arbitrarily to 1000px is weird. For a DE of 12 teams the line ends visually at the middle of the final match. 
Maybe setting the width of the line to the width of the bracket overview frame minus a static amount would work. I imagine that this would not dynamically update, but it's still a better solution that a static width in my opinion.

Reply:
I tried making it have the same width as the upperbracket, but that just returned 0. I can look for a another dynamic number.

Reply:
Yes, I ran into that as well. Maybe we should just create an issue instead - after all it's just a minor cosmetic thing.

Reply:
Wrote #59.

Reply:
d772c6d resolves #59